astropy/units/tests/test_quantity.py

```diff
@@ -667,6 +667,13 @@ def test_quantity_comparison(self):
 
         assert u.Quantity(1200, unit=u.meter) != u.Quantity(1, unit=u.kilometer)
 
+    def test_quantity_is_equivalent(self):
+        assert u.Quantity(10, unit='m').is_equivalent(u.Quantity(1, unit='cm'))
+        assert u.Quantity(10, unit='m').is_equivalent(u.pc)
+        assert u.Quantity(10, unit='kg').is_equivalent('g')
+        assert not u.Quantity(10, unit='m').is_equivalent(u.Quantity(1, unit='mag'))
+        assert not u.Quantity(10, unit='m').is_equivalent(u.arcsec)
+
```

Comment:
I would add a test passing through a unit with equivalencies, e.g., using the `u.spectral()` equivalency.